test/methods.jl

```diff
@@ -1,70 +1,46 @@
+
+        # The different floating point types to try and the associated required relative
+        # tolerance.
+        types = [Float32, Float64]
+
+        # The different functions to evaluate (`.f`), their first derivative at 1 (`.d1`),
+        # and second derivative at 1 (`.d2`).
+        foos = [
+            (f=sin, d1=cos(1), d2=-sin(1)),
+            (f=exp, d1=exp(1), d2=exp(1)),
+            (f=abs2, d1=2, d2=2),
+            (f=x -> sqrt(max(x + 1,0)), d1=0.5 / sqrt(2), d2=-0.25 / 2^(3/2)),
+        ]
+
+        # Test all combinations of the above settings, i.e. differentiate all functions using
+        # all methods and data types.
+        @testset "foo=$(foo.f), method=$m, type=$(T)" for foo in foos, m in methods, T in types
+            @testset "method-order=$order" for order in [1, 2, 3, 4, 5]
+                @test m(order, 0, adapt=2)(foo.f, T(1)) isa T
+                @test m(order, 0, adapt=2)(foo.f, T(1)) == T(foo.f(1))
+            end
 
-        # Test lower bound of `eps(T) / 1000`.
-        @test estimate_roundoff_error(x -> 1e-100, 0.0) == eps(1.0) / 1000
-        @test estimate_roundoff_error(x -> 1f-100, 0f0) == eps(1f0) / 1000
-    end
+            @test m(10, 1)(foo.f, T(1)) isa T
+            @test m(10, 1)(foo.f, T(1)) ≈ T(foo.d1)
 
-    # The different approaches to approximating the gradient to try.
-    methods = [forward_fdm, backward_fdm, central_fdm]
-
-    # The different floating point types to try and the associated required relative
-    # tolerance.
-    types = [Float32, Float64]
-
-    # The different functions to evaluate (.f), their first derivative at 1 (.d1),
-    # and second derivative at 1 (.d2).
-    foos = [
-        (f=sin, d1=cos(1), d2=-sin(1)),
-        (f=exp, d1=exp(1), d2=exp(1)),
-        (f=abs2, d1=2, d2=2),
-        (f=x -> sqrt(x + 1), d1=0.5 / sqrt(2), d2=-0.25 / 2^(3/2)),
-    ]
-
-    # Test all combinations of the above settings, i.e. differentiate all functions using
-    # all methods and data types.
-    @testset "foo=$(foo.f), method=$m, type=$(T)" for foo in foos, m in methods, T in types
-        @testset "method-order=$order" for order in [1, 2, 3, 4, 5]
-            @test m(order, 0, adapt=2)(foo.f, T(1)) isa T
-            @test m(order, 0, adapt=2)(foo.f, T(1)) == T(foo.f(1))
+            @test m(10, 2)(foo.f, T(1)) isa T
+            if T == Float64
+                @test m(10, 2)(foo.f, T(1)) ≈ T(foo.d2)
+            end
         end
+    end
 
-        @test m(10, 1)(foo.f, T(1)) isa T
-        @test m(10, 1)(foo.f, T(1)) ≈ T(foo.d1)
+    @testset "Test custom grid" begin
+        @test FiniteDifferenceMethod([-2, 0, 5], 1)(sin, 1) ≈ cos(1)
+    end
 
-        @test m(10, 2)(foo.f, T(1)) isa T
-        if T == Float64
-            @test m(10, 2)(foo.f, T(1)) ≈ T(foo.d2)
-        end
+    @testset "Test allocations" begin
+        m = central_fdm(5, 2, adapt=2)
+        @test (@benchmark $m(sin, 1)).allocs == 0
```

Comment:
To make this run more quickly, you could instead write something like:
```julia
@test (@benchmark $m(sin, 1); samples=1, evals=1).allocs == 0
```
This severly limits the number of function evaluations that `BenchmarkTools` is allowed to perform but doesn't seem to effect the allocation estimate too much.

Reply:
Good suggestion, thanks!